src/contracts/core/StrategyManager.sol

```diff
@@ -999,6 +889,11 @@ contract StrategyManager is
         );
     }
 
+    // @notice Returns 'true' if `staker` can undelegate and false otherwise
+    function stakerCanUndelegate(address staker) public view returns (bool) {
```

Comment:
I would prefer addresses only being able to be pod owners or only being able to be liquid restakers. We could check on deposit. I feel like this makes the state of the system smaller and easier to reason about for reasonable tradeoff?

Reply:
> I would prefer addresses only being able to be pod owners or only being able to be liquid restakers. We could check on deposit. I feel like this makes the state of the system smaller and easier to reason about for reasonable tradeoff?

IMO this is a major design decision, not something that should be made off-hand in a GitHub comment.

We would need to, at minimum:

- look at the current state to see if anyone currently falls into both categories, and make a plan on how to handle any such users
- add checks that enforce this behavior

Reply:
Agreed. Actually let's keep it this way. I like it less, but we cant stop people from restaking on beacon chain once they've already deposited LSTs.